optimize.py

```diff
@@ -158,9 +159,39 @@ def count_events(tree, cuts, eventWeightBranch):
   return events.size, np.sum(events).astype(float)
 
 #@echo(write=logger.debug)
-def get_significance(signal, bkgd, cuts, eventWeightBranch, insignificanceThreshold, bkgdUncertainty):
+def get_did(filename):
+  did_regex = re.compile('(\d{6,8})')
+  m = did_regex.search(filename)
+  if m is None: raise ValueError('Can\'t figure out the DID!')
+  return m.group(1)
+
+
+#@echo(write=logger.debug)
+def get_scaleFactor(filename):
+  did = get_did(filename)
+  weights = yaml.load(file(args.weightsFile))
+  weight = weights.get(did)
+  if weight is None:
+    raise KeyError("Could not find the weights for did=%s" % did)
+  scaleFactor = 1.0
+  scaleFactor /= weight.get('num events')
+  scaleFactor *= weight.get('cross section')
+  scaleFactor *= weight.get('filter efficiency')
+  scaleFactor *= weight.get('k-factor')
+  scaleFactor *= weights.get('global_luminosity') * 1000 #to account for units on luminosity
+  return scaleFactor
+  
+
+#@echo(write=logger.debug)
+def get_significance(signal, bkgd, cuts, eventWeightBranch, insignificanceThreshold, bkgdUncertainty, signal_scale, bkgd_scale):
   numSignal, weightedSignal = count_events(signal, cuts, eventWeightBranch)
   numBkgd, weightedBkgd = count_events(bkgd, cuts, eventWeightBranch)
+  
+  # apply scale factors and luminosity
+  numSignal = numSignal * signal_scale
```

Comment:
Don't scale numSignal? But that's what's fed into sig on line 208. Should weightedSignal by scaled and fed into the significance function?